[iOS] Use finer-grained media services when available

brentfulgham opened the pull request:
#### 6cbeaffa801faa58f3054b92e65a4ed8d5071773
<pre>
[iOS] Use finer-grained media services when available
<a href="https://bugs.webkit.org/show_bug.cgi?id=249033">https://bugs.webkit.org/show_bug.cgi?id=249033</a>
&lt;rdar://100435873&gt;

Reviewed by Per Arne Vollan.

Use more specific mach services for media features on platforms that support them.
This will allow us to narrow the scope to only those endpoints required for
web-related media tasks.

No change in behavior.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:

Canonical link: <a href="https://commits.webkit.org/257653@main">https://commits.webkit.org/257653@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cfd16b97f9c8ec8069e834fdadad2108fad05992

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99567 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8742 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32660 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108929 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169166 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/9328 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86047 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92041 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106846 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105333 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/7054 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/90574 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34017 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/88870 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21926 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/77286 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/90216 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2583 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23441 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/86104 "Built successfully") | 
| [⏳ 🛠 🧪 jsc-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-arm64-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2515 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/45835 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/28929 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/8663 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42916 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/88975 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5267 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4375 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/19899 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:
```diff
@@ -93,6 +93,32 @@
+            "com.apple.coremedia.mediaplaybackd.formatreader.xpc"
+            "com.apple.coremedia.mediaplaybackd.player.xpc"
+            "com.apple.coremedia.mediaplaybackd.remaker.xpc"
+            "com.apple.coremedia.mediaplaybackd.samplebufferaudiorenderer.xpc"
```
Are there other services we can remove access to when mediaaybackd is present? I may look like there are old counterparts to some of these new services?

pvollan replied:
Perhaps we can add telemetry to the old services if HAVE_MEDIAPLAYBACKD = 1?

brentfulgham replied:
@pvollan: Indeed, we should consider that in the future, but for now the relevant people working on this finer-grained XPC layer say they do not intend for us to remove the old items yet (though this will hopefully happen soon).

pvollan reviewed (approved):
R=me.

brentfulgham committed:
[iOS] Use finer-grained media services when available
https://bugs.webkit.org/show_bug.cgi?id=249033
<rdar://100435873>

Reviewed by Per Arne Vollan.

Use more specific mach services for media features on platforms that support them.
This will allow us to narrow the scope to only those endpoints required for
web-related media tasks.

No change in behavior.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:

Canonical link: https://commits.webkit.org/257653@main
